dbt_dagster_project/dbt_dagster_project/assets.py

```diff
@@ -0,0 +1,44 @@
+import json
+from typing import Any, Mapping
+
+from dagster import (AssetExecutionContext, DailyPartitionsDefinition,
+                     OpExecutionContext)
+from dagster_dbt import (DagsterDbtTranslator, DbtCliResource, dbt_assets,
+                         default_metadata_from_dbt_resource_props)
+
+from .constants import dbt_manifest_path
+
+
+@dbt_assets(manifest=dbt_manifest_path, exclude="fct_reviews")
+def dbtlearn_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
+    yield from dbt.cli(["build"], context=context).stream()
+
+
+# many dbt assets use an incremental approach to avoid
+# re-processing all data on each run
+# this approach can be modelled in dagster using partitions
+daily_partitions = DailyPartitionsDefinition(start_date="2022-01-24")
+
+class CustomDagsterDbtTranslator(DagsterDbtTranslator):
+    def get_metadata(self, dbt_resource_props: Mapping[str, Any]) -> Mapping[str, Any]:
+        metadata = {"partition_expr": "date"}
+        default_metadata = default_metadata_from_dbt_resource_props(dbt_resource_props)
+        return {**default_metadata, **metadata}
+
+
+@dbt_assets(manifest=dbt_manifest_path, 
+            select="fct_reviews",
+            partitions_def=daily_partitions,
+            dagster_dbt_translator=CustomDagsterDbtTranslator())
+def dbtlearn_partitioned_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
+    (
+        first_partition,
+        last_partition,
+    ) = context.asset_partitions_time_window_for_output(
```

Comment:
We use each partition key run and I see for the first time that there is actually a possibility to pack the backfill in one call. I liked that